src/test/java/test/listeners/InterceptorInvokeTwiceSimulateListener.java

```diff
@@ -0,0 +1,50 @@
+package test.listeners;
+
+import java.util.List;
+import org.testng.IMethodInstance;
+import org.testng.IMethodInterceptor;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+/**
+ * This listener is used to trigger the bug 1863: IMethodInterceptor will be invoked twice when
+ * listener implements both ITestListener and IMethodInterceptor via eclipse execution way
+ */
+public class InterceptorInvokeTwiceSimulateListener implements
+        ITestListener, IMethodInterceptor {
+
+    private int count = 0;
+
+    @Override
+    public void onStart(ITestContext context) {}
```

Comment:
all these blank methods as implementations from `ITestListener` can be removed now, because they are all default methods from `org.testng.ITestListener`. So we could just shorten this class as below

```java
public class InterceptorInvokeTwiceSimulateListener implements ITestListener, IMethodInterceptor {

  @Override
  public List<IMethodInstance> intercept(List<IMethodInstance> methods, ITestContext context) {
    //code goes here
    return methods;
  }
}
```

Reply:
Yes, this is a default method implementation in interface in JAVA8

Reply:
@wubin-ericsson - FYI.. TestNG has moved to using Java8. So you can refactor your listener, like how I shown it above.